lib/auth0/appEndpointHandler.js

```diff
@@ -6,11 +6,15 @@ module.exports = function (webtask, options, ctx, req, res, routingInfo) {
     function authenticate() {
         var token = options.getAccessToken(ctx, req);
         if (!token) {
+            var stateInfo = {
+                appPath: routingInfo.appPath,
+                query: routingInfo.query
+            };
             return options.loginError({
                 code: 401,
                 message: 'Unauthorized.',
                 error: 'Missing access token.',
-                redirect: routingInfo.baseUrl + '/login'
+                redirect: routingInfo.baseUrl + '/login?state=' + encodeURIComponent(JSON.stringify(stateInfo))
```

Comment:
Hi @scott-parsons, I spoke to @jaredhanson about this question. His recommendation was to store the redirect state in some form of client storage (cookie, localstorage) as a mapping of a unique, random _thunk_ to the state.

For example:

```js
{
    'adkd9242n22309jalk23': { path: '/...', ... }
}```

The _thunk_ value is then used as the `state` query parameter. When the flow redirects to the client upon auth completion, the client (browser) **must** ensure that it has a record of this _thunk_ and can then take appropriate action. Of course, encoding this state in a signed (and encrypted, ideally) JWT is also possible but ads complexity.